dev/set_matrix.py

```diff
@@ -419,6 +419,7 @@ def expand_config(config):
                         run=run,
                         package=pip_release,
                         version=ver,
+                        supported=Version(ver) <= Version(max_ver),
```

Comment:
@WeichenXu123 I renamed `newer_than_maximum` to `supported` (which I thin is clearer).